cesium/internal/domain/db.go

```diff
@@ -12,23 +12,34 @@ package domain
 import (
 	"context"
 	"github.com/synnaxlabs/alamos"
+	"github.com/synnaxlabs/cesium/internal/core"
 	"github.com/synnaxlabs/x/config"
 	"github.com/synnaxlabs/x/errors"
+	xio "github.com/synnaxlabs/x/io"
 	xfs "github.com/synnaxlabs/x/io/fs"
 	"github.com/synnaxlabs/x/override"
 	"github.com/synnaxlabs/x/query"
 	"github.com/synnaxlabs/x/telem"
 	"github.com/synnaxlabs/x/validate"
-	"io"
+	"sync/atomic"
 )
 
 var (
-	// ErrDomainOverlap is returned when a domain overlaps with an existing domain in the DB.
-	ErrDomainOverlap = errors.Wrap(validate.Error, "domain overlaps with an existing domain")
-	// RangeNotFound is returned when a requested domain is not found in the DB.
-	RangeNotFound = errors.Wrap(query.NotFound, "domain not found")
+	// ErrWriteConflict is returned when a domain overlaps with an existing domain in the DB.
+	ErrWriteConflict = errors.Wrap(validate.Error, "write conflicts with existing timerange in database")
+	// ErrRangeNotFound is returned when a requested domain is not found in the DB.
+	ErrRangeNotFound = errors.Wrap(query.NotFound, "timerange not found")
+	errDBClosed      = core.EntityClosed("domain.db")
 )
 
+func NewErrWriteConflict(tr1, tr2 telem.TimeRange) error {
+	return errors.Wrapf(ErrWriteConflict, "write overlaps on timerange %v", tr1.Intersect(tr2))
```

Comment:
I think this message could still be better -> "write overlaps with existing data occupying time range"